pennylane/_qubit_device.py

```diff
@@ -1186,6 +1199,83 @@ def mutual_info(self, wires0, wires1, log_base):
+        that are performed for each qubit and snapshot as an integer:
+
+        - 0 for Pauli X,
+        - 1 for Pauli Y, and
+        - 2 for PauliZ.
+
+        It also returns the measurement results (the ``bits``); 0 if the 1 eigenvalue
+        is sampled, and 1 if the -1 eigenvalue is sampled.
+
+        The device shots are used to specify the number of snapshots. If ``T`` is the number
+        of shots and ``n`` is the number of qubits, then both the measured bits and the
+        Pauli measurements have shape ``(T, n)``.
+
+        This implementation is device-agnostic and works by executing single-shot
+        tapes containing randomized Pauli observables. Devices should override this
+        if they can offer cleaner or faster implementations.
+
+        .. seealso:: :func:`~.classical_shadow`
+
+        Args:
+            wires (Sequence[int]): The wires to perform Pauli measurements on
+            n_snapshots (int): The number of snapshots
+            circuit (~.tapes.QuantumTape): The quantum tape that is being executed
+            seed (Union[int, None]): If provided, it is used to seed the random
+                number generation for generating the Pauli measurements.
+
+        Returns:
+            tensor_like[int]: A tensor with shape ``(2, T, n)``, where the first row represents
+            the measured bits and the second represents the recipes used.
+        """
+        if circuit is None:  # pragma: no cover
+            raise ValueError("Circuit must be provided when measuring classical shadows")
+
+        n_snapshots = self.shots
+
+        with set_shots(self, shots=1):
+            # slow implementation but works for all devices
+            n_qubits = len(self.wires)
+            device_wires = np.array(self.map_wires(wires))
+
+            if seed is not None:
+                # seed the random measurement generation so that recipes
+                # are the same for different executions with the same seed
+                rng = np.random.RandomState(seed)
+                recipes = rng.randint(0, 3, size=(n_snapshots, n_qubits))
+            else:
+                recipes = np.random.randint(0, 3, size=(n_snapshots, n_qubits))
+            obs_list = [qml.PauliX, qml.PauliY, qml.PauliZ]
+
+            outcomes = np.zeros((n_snapshots, n_qubits))
+
+            for t in range(n_snapshots):
+                # compute rotations for the Pauli measurements
+                rotations = [
+                    rot
+                    for wire in wires
+                    for rot in obs_list[
+                        recipes[t][self.wire_map[wire]]
+                    ].compute_diagonalizing_gates(wires=wire)
+                ]
+
+                self.reset()
+                self.apply(circuit.operations, rotations=circuit.diagonalizing_gates + rotations)
+
+                outcomes[t] = self.generate_samples()[0]
+
+        outcomes = outcomes[:, device_wires]
+        recipes = recipes[:, device_wires]
```

Comment:
```suggestion
                outcomes[t] = self.generate_samples()[0][wires]
```

Reply:
Seems like this would cause issues if `wires` is not a numpy array or tuple. How about
```
outcomes[t] = self.generate_samples()[0][mapped_wires]
```
where `mapped_wires` is the same as before?

Reply:
Nice catch!! 🎉